zivid_samples/CMakeLists.txt

```diff
@@ -14,35 +14,55 @@ catkin_package(
   CATKIN_DEPENDS zivid_camera
 )
 
-################
-## C++ Sample ##
-################
-
-set(SAMPLE_CPP_NAME "sample_capture_cpp")
-add_executable(${SAMPLE_CPP_NAME} src/sample_capture.cpp)
-add_dependencies(${SAMPLE_CPP_NAME} ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
-target_include_directories(${SAMPLE_CPP_NAME}
-  SYSTEM PRIVATE
-  ${catkin_INCLUDE_DIRS}
-)
-target_link_libraries(${SAMPLE_CPP_NAME}
-  ${catkin_LIBRARIES}
-)
+#################
+## C++ Samples ##
+#################
 
-install(TARGETS ${SAMPLE_CPP_NAME}
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-)
+function(register_cpp_sample)
+  cmake_parse_arguments(
+    ARG
+    ""
+    "NAME;SRC"
+    ""
+    ${ARGN}
+  )
+  add_executable(${ARG_NAME} ${ARG_SRC})
+  add_dependencies(${ARG_NAME} ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
+  target_include_directories(${ARG_NAME}
+    SYSTEM PRIVATE
+    ${catkin_INCLUDE_DIRS}
+  )
+  target_link_libraries(${ARG_NAME}
+    ${catkin_LIBRARIES}
+  )
+  install(TARGETS ${ARG_NAME}
+    ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+    LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+    RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+  )
+endfunction()
 
-###################
-## Python Sample ##
-###################
+register_cpp_sample(NAME sample_capture_cpp SRC src/sample_capture.cpp)
 
-install(PROGRAMS
-  scripts/sample_capture.py
-  DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-)
+####################
+## Python Samples ##
+####################
+
+function(register_python_sample)
+  cmake_parse_arguments(
+    ARG
+    ""
+    "SRC"
+    ""
+    ${ARGN}
+    )
+  install(PROGRAMS
```

Comment:
I have also seen `catkin_install_python(PROGRAMS scripts/<myScript> DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})` being used. Are you aware if there is any difference?

Reply:
Based on /opt/ros/melodic/share/catkin/cmake/catkin_install_python.cmake this seems to do some additional stuff like adding a shebang line which we already have, some stuff for installing it on Python etc. I suggest to leave this as is for now, until we get complaints.